src/components/WeatherDetails/index.js

```diff
@@ -0,0 +1,37 @@
+import React from "react";
+
+import "./styles.scss";
+import SearchBar from "../SearchBar";
+import Details from '../Details'
+import SearchProgress from "../SearchProgress";
+import InfoBox from "../InfoBox"
+import {
+  Switch,
+  Route,
+} from "react-router-dom";
+
+const WeatherDetails = () => {
+  return (
+    <div className="container weather-details">
+        
+        <SearchProgress/>
+        <Switch >
+          <Route path="/" exact>
+          <SearchBar/>
+          <InfoBox/>
+          </Route>
+          <Route path="/city/:cityId">
```

Comment:
City name should not be a part of the url router. Either pass the city name as query param or use id instead